framework/test/gtest/ServiceHooksTest.cpp

```diff
@@ -305,6 +292,54 @@ namespace
 
 } // end unnamed namespace
 
+class MockServiceEventListenerHook : public ServiceEventListenerHook
+{
+  public:
+    MockServiceEventListenerHook(int id, BundleContext const& context)
+        : fake_(TestServiceEventListenerHook(id, context))
+    {
+    }
+
+    using MapType = ShrinkableMap<BundleContext, ShrinkableVector<ServiceListenerHook::ListenerInfo>>;
+
+    MOCK_METHOD(void, Event, (ServiceEvent const& event, MapType& listeners), (override));
+
+  private:
+    TestServiceEventListenerHook fake_;
+};
+
+class MockServiceFindHook : public ServiceFindHook
+{
+  public:
+    MockServiceFindHook(BundleContext const& context)
+        : fake_(TestServiceFindHook(context))
+    {
+    }
+
+    MOCK_METHOD(void,
+                Find,
+                (BundleContext const& context,
+                 std::string const& name,
+                 std::string const& filter,
+                 ShrinkableVector<ServiceReferenceBase>& references),
+                (override));
+
+  private:
+    TestServiceFindHook fake_;
+};
+
+class MockServiceListenerHook : public ServiceListenerHook
+{
+  public:
+    MockServiceListenerHook(BundleContext const& context) : fake_(TestServiceListenerHook(context)) {}
```

Comment:
I don't think this is correct. The `fake_` member is not needed for the mock and for testing the ordering of callbacks.

Reply:
I guess you're right. We no longer need the `fake_` variable so we don't need to worry about id. I was thinking we still needed to have the fake_ variable so we needed the id, because the `TestServiceEventListener` requires the id for more than just ordering so it needed to remain in its constructor and therefore the constructor of the mock as well, but it doesn't need to stay in the mock if we don't have the `fake_`.